src/coreclr/src/ToolBox/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -478,6 +478,15 @@ void MethodContext::dumpToConsole(int mcNumber)
     {
         printf(" method context #%d", mcNumber);
     }
+
+    // Dump method name, etc., to output.
+    char bufferIdentityInfo[METHOD_IDENTITY_INFO_SIZE];
+    int cbLen = dumpMethodIdentityInfoToBuffer(bufferIdentityInfo, METHOD_IDENTITY_INFO_SIZE);
+    if (cbLen >= 0)
+    {
+        printf(" %s", bufferIdentityInfo);
```

Comment:
Might be better to use `" %.*s"` instead: `printf(" %.*s", cbLen, bufferIdentityInfo)`.  (`snprintf_s()`, as used by `dumpMethodIdentityInfoToBuffer()`, takes in the size of the buffer (`METHOD_IDENTITY_INFO_SIZE`), but returns the number of characters there, without the terminating `NUL`.)

Reply:
why do you think that it is better? `dumpMethodIdentityInfoToBuffer` adds a null character at the end (it is index is `cbLen`), so `printf(" %s", bufferIdentityInfo);` will work as expected.

Reply:
snprintf_s() will return an error if the terminating NUL won't fit in the buffer, and that should be fixed in dumpMethodIdentifyInfoToBuffer().  (In the unlikelyhood the buffer is too big, `buff` will go backwards, etc.)

Reply:
Yes, there's definitely a potential problem because we don't check for error conditions on sprintf_s. The buffer is 64KB, so it seems unlikely...

Reply:
The buffer is big enough, however if it fails the first time snprintf_s() is called, the pointer will point outside of the buffer.